internal/armadactl/submit.go

```diff
@@ -14,44 +14,39 @@ import (
 
 // Submit a job, represented by a file, to the Armada server.
 // If dry-run is true, the job file is validated but not submitted.
-func (a *App) Submit(path string, dryRun bool) (outerErr error) {
+func (a *App) Submit(path string, dryRun bool) error {
 
 	ok, err := validation.ValidateSubmitFile(path)
 	if !ok {
-		fmt.Fprintf(a.Out, "Invalid jobfile: %s/n", err)
-		return nil
+		return err
```

Comment:
I am a big fan of using `errors.Wrap` from `github.com/pkg/errors` to provide additional context to errors.
i.e. `errors.Wrap(err, "error validating submited file")`, it is especially useful if the errors in `validation.ValidateSubmitFile` do not give info on the calling context.

Reply:
I don't think we should add a stack trace here; the stack trace should be added where the error first occurs. This code only annotates the error with some additional info.